### PR TITLE
Benchpress micro

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
Wow, docs for benchpress! Awesome @tbosch !

**Comment by rkirov** [comment]
LGTM.

File: modules/benchpress/docs/index.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+Benchpress is a framework for e2e performance tests.
+
+# Why?
+
+There are so called "micro benchmarks" that esentially use a stop watch in the browser to measure time
+(e.g. via `performance.now()`). This approach is limited to time, and in some cases memory
+(Chrome with special flags), as metric. It does not allow to measure:
+
+- rendering time: e.g. the time the browser spends to layout or paint elements. This can e.g. used to
```
Comment (gkalpak) [inline modules/benchpress/docs/index.md]: _used_  -->  _be used_

File: modules/benchpress/docs/index.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+Benchpress tries to fill this gap, i.e. allow to access all kinds of performance metrics in an automated way.
+
+
+# How it works
+
+Benchpress uses webdriver to read out the so called "performance log" of browsers. This contains all kinds of interesting
+data, e.g. when a script started/ended executing, gc started/ended, the browser painted something to the screen, ...
+
+As browsers are different, benchpress has plugins to normalizes these events.
```
Comment (gkalpak) [inline modules/benchpress/docs/index.md]: _normalizes_  -->  _normalize_

File: modules/benchpress/docs/index.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+```
+
+A benchmark driver could look like this:
+
+```
+// A runner contains the shared configuration
+// and can be shared across multiple tests.
+var runner = new Runner(...);
+
+driver.get('http://myserver/index.html');
```
Comment (gkalpak) [inline modules/benchpress/docs/index.md]: Where does the driver come from ?

File: modules/benchpress/docs/index.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+If the application under test would like to, it can measure on its own.
+E.g.
+
+```
+index.html:
+
+<button id="measure" onclick="measure()">Measure document.createElement</button>
+<script>
+  function measure() {
+    console.time('createElement*10000');
```
Comment (gkalpak) [inline modules/benchpress/docs/index.md]: _10000_  -->  _100000_ ?

File: modules/benchpress/docs/index.md [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+    for (var i=0; i<100000; i++) {
```
Comment (gkalpak) [inline modules/benchpress/docs/index.md]: _100000_  -->  _10000_

**Pull request closed without merging**